opal/mca/base/mca_base_var.c

```diff
@@ -1584,10 +1602,16 @@ int mca_base_var_register_synonym (int synonym_for, const char *project_name,
 
 static int var_get_env (mca_base_var_t *var, const char *name, char **source, char **value)
 {
-    char envvar[128];
+    const int max_len = sizeof(OPAL_MCA_PREFIX) +
+        MCA_BASE_MAX_PROJECT_NAME_LEN +
+        MCA_BASE_MAX_TYPE_NAME_LEN +
+        MCA_BASE_MAX_COMPONENT_NAME_LEN +
+        MCA_BASE_MAX_VARIABLE_NAME_LEN +
+        4 + 1; // 4=max number of _'s, 1=trailing \0
+    char envvar[max_len];
     int ret;
 
-    ret = snprintf(envvar, 128, "%s%s", mca_prefix, name);
+    ret = snprintf(envvar, max_len, "%s%s", mca_prefix, name);
```

Comment:
you need to change the snprintf below as well. And there you have to account for the "SOURCE_" additional string.